### DISCUSSION TITLE
.NET Aspire with Pulumi ?

### DISCUSSION THREAD

**Discussion by VincentH-Net** [pulumi/pulumi#15015] [Ideas]
What are Pulumi (community)'s thoughts on combining [.NET Aspire](https://learn.microsoft.com/en-us/dotnet/aspire/get-started/aspire-overview) with Pulumi ?

# Context
I am designing a distributed .NET & Azure application; for IaC I am considering Pulumi. The application uses .NET Aspire to simplify distributed application development.

Since .NET Aspire also automates deployment to a degree (in my case of Microsoft Orleans to Azure Container Apps) [via AZD](https://learn.microsoft.com/en-us/dotnet/aspire/deployment/azure/aca-deployment-azd-in-depth?tabs=windows), I am wondering if / how these two tools could / should be used together?

I'm interested to hear thoughts / insights of Pulumi team as well as community - thanks!

**Comment by VincentH-Net** [comment]
Possibly this could be realized by [supporting Pulumi from AZD](https://github.com/Azure/azure-dev/issues/93)

**Comment by Zaid-Ajaj** [comment]
Hi @VincentH-Net I've looked into Aspire and what it does for IaC deployments. In theory it is possible to generate Pulumi IaC code from the aspire manifest definition, that will require a new converter plugin. I've also seen that you could generate IaC code in the form of Bicep and take it from there. If you have Bicep code, you could use https://github.com/pulumi/pulumi-converter-bicep to get Pulumi code from it but from what I've seen it might not look as clean as if we build a specialized converter for aspire specifically.

**Reply by VincentH-Net** [reply]
A specialized converter is something pulumi could do - however, looking at [Generate Bicep from .NET Aspire app model](https://learn.microsoft.com/en-us/dotnet/aspire/deployment/azure/aca-deployment-azd-in-depth?branch=main&tabs=windows#generate-bicep-from-net-aspire-app-model) it seems the Bicep files that Aspire generates could be  converted using https://github.com/pulumi/pulumi-converter-bicep

@Zaid-Ajaj did you perchance try that (or know of someone who did?)

**Reply by tomachristian** [reply]
Why generate anything when a Pulumi program could take as an input the manifest directly?

**Comment by atrauzzi** [comment]
What about if you're not running on Azure?
